source/h2_connection.c

```diff
@@ -1777,6 +1871,13 @@ static void s_cross_thread_work_task(struct aws_channel_task *task, void *arg, e
         }
     }
 
+    /* Move pending PING to thread data */
+    if (!aws_linked_list_empty(&pending_ping)) {
+        while (!aws_linked_list_empty(&pending_ping)) {
```

Comment:
We can probably just push the front of pending_ping to the end of the pending_ping_queue, right? It will be more efficient.

Reply:
If performance will not matter for our use case, let's go with the existing one for now, I may add those helpers after the window_update.